server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -417,6 +417,10 @@ protected void doXContentBody(XContentBuilder builder, Params params) throws IOE
 
     protected abstract String contentType();
 
+    protected boolean canParseObjects() {
+        return false;
```

Comment:
**Note for reviewers:**
By reading public docs the only Fields that can parse objects are Percolator and AbstractGeometry (that covers geo_point, geo_shape, point, shape). Am I missing other Fields?

Reply:
There's more: I would look at the mappers that override `FieldMapper#parse(DocumentParserContext context)` as opposed to `parseCreateField`. Those are the ones that support complex formats, and out of those you'd need to double check which ones support objects (e.g. does the parsing code handle objects?)